activesupport/lib/active_support/cache.rb

```diff
@@ -18,6 +18,65 @@ module Cache
     autoload :NullStore,        "active_support/cache/null_store"
     autoload :RedisCacheStore,  "active_support/cache/redis_cache_store"
 
+
+    class KeyMaster
+      attr_reader :cache_key, :cache_version
+
+      def initialize(key = nil)
+        @cache_key = String.new
+        @cache_version = String.new
+        self << key
+      end
+
+      def <<(key)
+        return if key.nil?
+        if key.respond_to?(:cache_key)
+          store_key(key.cache_key.to_s)
+          store_version(key)
+          return
+        end
+
+        case key
+        when Array
+          key.each do |k|
+            self << k
+          end
+        when Hash
+          key = key.to_a
+          store_version(key)
+          # handle expand version here
+          key.sort_by! { |k, _| k.to_s }
+          key.collect! { |k, v| "#{k}=#{v}" }
+          key.each do |k|
+            append_part(key)
+          end
+        else
+          store_version(key)
+          store_key(key)
+        end
+      end
+
+      private def store_key(key)
```

Comment:
please don't use this style in the rails codebase. 🙏

Reply:
Thank you for the review, I will change it.

I will also tell you why I prefer this style. 

Since many rails classes are very long, it is hard to know what method is public and private while you are reading the code, especially if it has no documentation and isn't explicitly `:nodoc:`, which most code isn't.

It looks like it is possible to add a rule to rubocop to detect this issue https://rubocop.readthedocs.io/en/latest/cops_style/#styleaccessmodifierdeclarations. Should it be added to the rules?

Reply:
Good idea. :+1: for that rule.